### PR TITLE
Natural booze

### PR DESCRIPTION
**Pull request by MTandi**
## About The Pull Request

Makes fermented booze take quality and power from the endurance and lifespan stats of the fruit.
These are the most neglected plant stats, and usually plants that have these high by default, do not have much nutriments.

This change makes fermented drinks a bit more desirable because your effort is reflected in their stats. And you get better mood buff by making them higher.

Fruit wine and other reagents get a "Natural" prefix when made through fermentation.

image

I consider turning it into a more general solution so that dispensed drinks will also have purity and if the barman wants to aim for an absolute quality cocktail, he may want to cooperate with botany like chef does.

## Why It's Good For The Game

Fermented drinks are now a bit special compared to the free booze dispenser variants.

## Changelog

:cl:
add: Fermented drinks give mood buffs according to the stats of fermented fruit
/:cl:

### CODE REVIEW

**Comment by CRITAWAKETS** [comment]
this seems like the maximum booze power is just 100%, and i am now disappoint

**Comment by MTandi** [comment]
> this seems like the maximum booze power is just 100%, and i am now disappoint

I was just not sure what it can lead to if I allow it above the original values. Will consider allowing creating extra strong vodka that would screw your mind with extra strong potatoes later.

**Comment by Farquaar** [comment]
Does this apply to freshly-squeezed juice as well?

**Comment by MTandi** [comment]
> Does this apply to freshly-squeezed juice as well?

Only distill reagents and fruit wine. Juice is in the further plans: https://hackmd.io/@ZSLHD7nTTlWNegwCVVdJBw/ryqRgDluj

**Comment by Overuseddart** [comment]
Ooh this looks cool, maybe we can make special effects for good brewin.

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 1]
```diff
@@ -37,7 +37,13 @@ All effects don't start immediately, but rather get worse over time; the rate is
 91-100: Dangerously toxic - swift death
 */
 
-/datum/reagent/consumable/ethanol/New()
+/datum/reagent/consumable/ethanol/New(list/data)
+	if(data)
```
Comment (Time-Green) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:41]: LAZYLEN?

**Review by Time-Green — CHANGES_REQUESTED** [review]
Looks good! Right now it's a small nerf, but 'natural' is pretty cool and we can add 200% purity turbo vodka later

**Comment by MTandi** [comment]
> Looks good! Right now it's a small nerf, but 'natural' is pretty cool and we can add 200% purity turbo vodka later

It's not a nerf, since distill reagents like wine and beer didn't have any quality at all. 
And for growables it will hardly be 0, since all plants have some lifespan and endurance to begin with.

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm (hunk 1 above)
Comment (MTandi) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: I have no clue what's the difference, but sure.

Comment (Time-Green) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: It also checks if your list isnt empty

**Comment by Time-Green** [comment]
> > Looks good! Right now it's a small nerf, but 'natural' is pretty cool and we can add 200% purity turbo vodka later
> 
> It's not a nerf, since distill reagents like wine and beer didn't have any quality at all. And for growables it will hardly be 0, since all plants have some lifespan and endurance to begin with.

Whatever we ball

**Pull request merged**